contracts/src/v0.8/functions/dev/v1_0_0/FunctionsBilling.sol

```diff
@@ -203,11 +203,16 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
       revert InsufficientBalance();
     }
 
-    bytes32 requestId = _computeRequestId(
-      address(this),
-      request.requestingContract,
-      request.subscriptionId,
-      request.initiatedRequests + 1
+    uint32 timeoutTimestamp = uint32(block.timestamp + config.requestTimeoutSeconds);
+    bytes32 requestId = keccak256(
+      abi.encode(
+        address(this),
+        request.requestingContract,
+        request.subscriptionId,
+        request.initiatedRequests + 1,
+        estimatedTotalCostJuels,
+        timeoutTimestamp
```

Comment:
This shouldn't be a blocker for this PR but I want to point out that there is a scenario where contract config is changed at roughly the same time as a request is built and when these two events get re-orged into a different order then some other fields might not match (adminFee, donFee, overheads).

Can we check how expensive it is to add all the fields here? Are there any downsides?

Reply:
Also do we know how our Frontend is going to behave with a request ID that changes on re-org? How about our dashboard and alerting?

Reply:
Based on recent discussion, we'll want to add tx.origin and request.data here as well.